src/main/java/jenkins/plugins/git/GitSCMSource.java

```diff
@@ -456,35 +467,20 @@ public FormValidation doCheckCredentialsId(@AncestorInPath Item context,
             }
 
             value = Util.fixEmptyAndTrim(value);
-            if (value == null) {
-                return FormValidation.ok();
-            }
 
             remote = Util.fixEmptyAndTrim(remote);
             if (remote == null)
             // not set, can't check
             {
                 return FormValidation.ok();
             }
+            StandardCredentials credentials = value == null ? null :
+                    CredentialsMatchers.firstOrNull(CredentialsProvider
+                                    .lookupCredentials(StandardCredentials.class, context, ACL.SYSTEM,
```

Comment:
❌ I just looked quickly at the code, no time for deeper investigation right now.

It seems that the removal of 
```
context instanceof Queue.Task	
? Tasks.getAuthenticationOf((Queue.Task) context)	
: ACL.SYSTEM
```
 means that you do not support the `authorize-project` plugin as you do not respect the potential authorization restriction put in place and could be a security vulnerability in such case.